core/services/relay/evm/evm.go

```diff
@@ -458,7 +459,13 @@ type configTransmitterOpts struct {
 	pluginGasLimit *uint32
 }
 
+// newOnChainContractTransmitter creates a new contract transmitter with the default queueing topic.
 func newOnChainContractTransmitter(ctx context.Context, lggr logger.Logger, rargs commontypes.RelayArgs, transmitterID string, ethKeystore keystore.Eth, configWatcher *configWatcher, opts configTransmitterOpts, transmissionContractABI abi.ABI) (*contractTransmitter, error) {
```

Comment:
This style of extended constructors does not scale. How about using a pointer instead?
And where is the term "topic" coming from? `NewQueueingTxStrategy` takes a `subject uuid.UUID`. Let's keep the names aligned.
```suggestion
// newOnChainContractTransmitter creates a new contract transmitter.
// If subject is nil, rargs.ExternalJobID will be used instead.
func newOnChainContractTransmitter(ctx context.Context, lggr logger.Logger, rargs commontypes.RelayArgs, transmitterID string, ethKeystore keystore.Eth, configWatcher *configWatcher, opts configTransmitterOpts, transmissionContractABI abi.ABI, subject *uuid.UUID) (*contractTransmitter, error) {
```

Reply:
> How about using a pointer instead?

Could you demonstrate what you mean with an short example?

Reply:
I would prefer not to modify the function signature because this would require updating all the call sites and will create unnecessary cognitive load on the person merging `chainlink` into `ccip` next time. Will you be OK with using the option pattern?

Reply:
Sounds good